cmd/esbulk/esbulk.go

```diff
@@ -114,136 +78,101 @@ func main() {
 		password = parts[1]
 	}
 
-	options := esbulk.Options{
-		Servers:    serverFlags,
-		Index:      *indexName,
-		DocType:    *docType,
-		BatchSize:  *batchSize,
-		Verbose:    *verbose,
-		Scheme:     "http",
-		MaxRetries: *maxRetries,
-		IDField:    *idfield,
-		Username:   username,
-		Password:   password,
+	defaultOptions := esbulk.Options{
+		Servers:     serverFlags,
+		Index:       *indexName,
+		Purge:       *purge,
+		Mapping:     *mapping,
+		NumWorkers:  *numWorkers,
+		ZeroReplica: *zeroReplica,
+		GZipped:     *gzipped,
+		DocType:     *docType,
+		BatchSize:   *batchSize,
+		Verbose:     *verbose,
+		Scheme:      "http",
+		IDField:     *idfield,
+		Username:    username,
+		Password:    password,
+		MaxRetries:  *maxRetries,
 	}
 
-	if *verbose {
-		log.Println(options)
-	}
+	counter := 0
+	start := time.Now()
+	var reader io.Reader
 
-	if *purge {
-		if err := esbulk.DeleteIndex(options); err != nil {
-			log.Fatal(err)
+	if *sourceDir != "" {
+		// process files from source directory
+		files, err := ioutil.ReadDir(*sourceDir)
+		if err != nil {
+			log.Fatalf("failed to list directory due to %s", err)
 		}
-		time.Sleep(5 * time.Second)
-	}
 
-	// Create index if not exists.
-	if err := esbulk.CreateIndex(options); err != nil {
-		log.Fatal(err)
-	}
+		for _, file := range files {
+			if file.IsDir() {
+				continue
+			}
 
-	if *mapping != "" {
-		var reader io.Reader
-		if _, err := os.Stat(*mapping); os.IsNotExist(err) {
-			reader = strings.NewReader(*mapping)
-		} else {
-			file, err := os.Open(*mapping)
+			path := filepath.Join(*sourceDir, file.Name())
+			options, err := esbulk.ParseLDJFilename(path, defaultOptions)
 			if err != nil {
-				log.Fatal(err)
+				log.Print(err)
+				continue
```

Comment:
@nkly for our use-case, the directory only contains the LDJ files that should be processed. My concern was that if the files are not named correctly (e.g. due to a bug in the filename generation), it would be better to crash as compared to silently proceeding. I think its reasonable to expect that the `-dir` arg points to a directory with only LDJ files that follow the naming scheme.

Reply:
@wmalik Sure. I just had the impression that these changes are gonna be merge back to the original repo; in that case you can't guarantee that it won't be executed on a system that creates hidden system files, i.e. OS X. If they stay only in our fork, then it makes sense to crash.

Reply:
@wmalik @nkly For example in our use case we have non LDJ files in output directory. What about explicitly checking for `.ldj`  and fail only if it was not possible to parse file with that extensions? WDYT?

Reply:
@tgrk @tgrk Sorry my mistake, we do have non-ldj files in that directory. 

> What about explicitly checking for .ldj and fail only if it was not possible to parse file with that extensions?

^This sounds great :+1:

Reply:
> @wmalik Sure. I just had the impression that these changes are gonna be merge back to the original repo; in that case you can't guarantee that it won't be executed on a system that creates hidden system files, i.e. OS X. If they stay only in our fork, then it makes sense to crash.

Good point. I concede.